examples/sim_fibo_ad_cp.cpp

```diff
@@ -206,6 +207,38 @@ try
                                   *props, deck, gravity[2], state);
     }
 
+    std::shared_ptr<BlackoilState> distributed_state=Dune::stackobject_to_shared_ptr(state);
+    std::shared_ptr<Opm::BlackoilPropsAdFromDeck> distributed_props=new_props;
+    Dune::CpGrid distributed_grid = *grid;
+    // At this point all properties and state variables are correctly initialized
+    // If there are more than one processors involved, we now repartition the grid
+    // and initilialize new properties and states for it.
+    if( grid->comm().size()>=1 )
+    {
+        if(!param.getDefault("output_vtk", true))
+        {
+            OPM_THROW(std::logic_error, "We only support vtk output during parallel runs");
+        }
+        grid->loadBalance();
```

Comment:
The grid holds  an instance of the global and one of the distributed CpGridData as shared_ptrs. The copy is shallow as it does only copy the pointers. In addition theres is a pointer that points to the current CpGridData used. This can be changed with the switchTo... statements. Basically we need one representation with the global grid and one with the distributed one for scattering the data to the distributed case. BTW: the switchToDistributedView() is superfluous but there to make clear what happens.